api/v1alpha1/toolchainconfig_types.go

```diff
@@ -244,6 +244,18 @@ type RegistrationServiceConfig struct {
 	// Keeps parameters necessary for the registration service verification config
 	// +optional
 	Verification RegistrationServiceVerificationConfig `json:"verification,omitempty"`
+
+	// UICanaryDeploymentWeight specifies the threshold of users that will be using the new UI.
+	// This configuration option is just a temporary solution for rolling out our new RHDH based UI using canary deployment strategy.
+	// Once we switch all our users to the new UI this will be removed.
+	// How this works:
+	// - backend returns a weight
+	// - old UI assigns a sticky random number for each user
+	// - if the user has a number within the weight returned from the backend than user get's redirect to to new UI
+	// - if the user has a number above the weight they keep using the current UI
+	//
+	// +optional
```

Comment:
just a detail, but you can add a validation of the input
```suggestion
	// +optional
	// +kubebuilder:validation:Minimum=0
	// +kubebuilder:validation:Maximum=10
	// +kubebuilder:default=10
```

Reply:
+1

Reply:
Yeah, tbh I'm not sure what range we want to use.

I though about 1 - 100.

Reply:
I've added the range (0 - 100 ) , but not the default, since that is defined in the registration service helper function.

Reply:
Yes, 1-100 is preferable IMO.